pkg/apis/operator/v1alpha1/apiserver_types.go

```diff
@@ -24,9 +24,11 @@ import (
 // APIServerSpec defines the desired state of APIServer
 type APIServerSpec struct {
 	// TLSSecretRef references the TLS certificate and private key for serving the KubeCarrier API.
-	TLSSecretRef ObjectReference `json:"tlsSecretRef"`
+	// +optional
+	TLSSecretRef *ObjectReference `json:"tlsSecretRef,omitempty"`
 	// OIDC specifies OpenID Connect configuration for API Server authentication
-	OIDC APIServerOIDCConfig `json:"oidc"`
+	// +optional
+	OIDC *APIServerOIDCConfig `json:"oidc,omitempty"`
```

Comment:
Could you add here `AnonymousAuthConfig` pointer field?  I'd love allowing user to include the Anonymous auth if they wish to, orthogonal whether the OIDC is included or not.  Thus something like this:

```yaml
    anonymousAuth: 
        enabled: true
```

Possible extension in  the future:

```yaml
    anonymousAuth: 
        enabled: true
        user: ...
        group: [] 
```

Thus in the resource creation, you check for all auth providers with non-nil config and include them in the API server setup. The ordering should be "OIDC,Anonymous";  and maybe in the future, we could allow users to set the ordering explicitly. 

At the end of the day, the user should be empowered picking and choose which auth providers he's going to use and whether he wants anonymous auth enabled or not. 

After this is implemented I'll LGTM it.

P.S.
alternative YAML schema:

```
authentication:
- oidc: {...}
- anonymous: {enabled: true} 
```

and this would also configure auth provider ordering explicitly by the user, and not our own heuristics. 

@thetechnick

Reply:
I like the idea, but I would like to see this done in another PR.
(this one is already fairly large, due to all the boilerplating)

Reply:
https://github.com/kubermatic/kubecarrier/issues/470